### DISCUSSION TITLE
Support Mirrors for case classes with implicits

### DISCUSSION THREAD

**Discussion by dwijnand** [scala/scala3#17891] [Feature Requests]
## Minimized code

```scala
case class WithImplicit1(pos: Int, text: String)(implicit x: Numeric[Int]) { def x1 = x.one }
```

## Output

```scala
scala> import scala.compiletime._, scala.deriving._

scala> val m = summon[Mirror.Of[WithImplicit1]]
1 |val m = summon[Mirror.Of[WithImplicit1]]
  |                                        ^
  |no implicit argument of type deriving.Mirror.Of[WithImplicit1] was found for parameter x of method summon in object Predef
```

## Expectation

I expect a `Mirror.Product` exists for such a class.

**Comment by bishabosha** [comment]
linkind to lampepfl/dotty#9985 which was closed in favour of status quo

**Comment by bishabosha** [comment]
I believe that mirror product only supports a single parameter list due to the need to implement the `Mirror.Product.fromProduct` method which is implemented based on the shape of that single parameter list.

Here the specification requires a single parameter list:

https://github.com/lampepfl/dotty/blob/609d8d6bdbe9be33080967cdac478a6957237356/compiler/src/dotty/tools/dotc/transform/SymUtils.scala#L81

**Comment by Lasering** [comment]
Related: lampepfl/dotty#17777
